### PR TITLE
Add generation of cluster autoscaler addon's manifest

### PR DESCRIPTION
**Pull request by marccarre**
### Description

This PR essentially adds:
```go
func GenerateManifests(params TemplateParameters) ([]byte, error)
```
under `pkg/addons/clusterautoscaler`, which can be used to generate the manifests required to easily [install the cluster autoscaler on EKS](https://aws.amazon.com/premiumsupport/knowledge-center/eks-cluster-autoscaler-setup/).

Note that, it relies on a YAML Go template which is embedded in the `eksctl` binary using `vfsgen`, added to the dependencies via:
```bash
$ GO111MODULE=on go get -u github.com/shurcooL/vfsgen
$ GO111MODULE=on go mod vendor
```

### Checklist

- [x] Code compiles correctly (i.e `make build`)
- [x] Added tests that cover your change (if possible)
- [x] All unit tests passing (i.e. `make test`)
- [x] Added/modified documentation as required (such as the `README.md`, and `examples` directory)
- [x] ~Manually tested~

### CODE REVIEW

**Comment by marccarre** [comment]
N.B.: given we want to use an abstraction for templates' processing, this PR now depends on #1132. Once that PR is merged, I'll update _this one_ to use that abstraction.

File: pkg/addons/clusterautoscaler/cluster_autoscaler.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+// GenerateManifests fills the cluster autoscaler's YAML templates with the
+// provided template parameters to generate actionable YAML manifests.
+func GenerateManifests(params TemplateParameters) ([]byte, error) {
+	if err := params.Validate(); err != nil {
+		return nil, err
+	}
+	manifests := [][]byte{}
+	err := vfsutil.WalkFiles(Templates, "/", func(path string, info os.FileInfo, rs io.ReadSeeker, err error) error {
+		if err != nil {
+			return fmt.Errorf("cannot walk embedded files: %s", err)
```
Comment (rndstr) [inline pkg/addons/clusterautoscaler/cluster_autoscaler.go]: nit: may also use `errors.Wrap()`

File: pkg/addons/clusterautoscaler/cluster_autoscaler_test.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
+package clusterautoscaler_test
+
+import (
+	"testing"
```
Comment (errordeveloper) [inline pkg/addons/clusterautoscaler/cluster_autoscaler_test.go]: Did you consider using ginko/gomega? That's what all other tests use...

File: pkg/addons/clusterautoscaler/templates_generate.go [hunk 3]
```diff
@@ -0,0 +1,21 @@
+// +build ignore
+
+package main
+
+import (
+	"log"
+
+	"github.com/shurcooL/vfsgen"
```
Comment (errordeveloper) [inline pkg/addons/clusterautoscaler/templates_generate.go]: We use bindata in other place, is there any particular reason to use this package here?

Comment (marccarre) [reply pkg/addons/clusterautoscaler/templates_generate.go]: Didn't know about `bindata`. I used `vfsgen` as that's what is used in Flux and in WKS.

File: pkg/addons/clusterautoscaler/cluster_autoscaler_test.go (hunk 2 above)
Comment (marccarre) [reply pkg/addons/clusterautoscaler/cluster_autoscaler_test.go]: Sure, will re-write... 😓

File: pkg/addons/clusterautoscaler/cluster_autoscaler.go (hunk 1 above)
Comment (marccarre) [reply pkg/addons/clusterautoscaler/cluster_autoscaler.go]: Good point. (FWIW, I took this from Flux, and didn't pay too much attention to such details.)

**Comment by martina-if** [comment]
I think that if the other components are in a base repo (today "eks-gitops-example") this ones should also be in that quickstart repo as yaml templates. See https://github.com/weaveworks/eks-gitops-example/pull/3#event-2545776922

I don't understand why tehse are special and have to be  embedded in eksctl. Is it because it needs different variables? We can incorporate those in the TemplateParameters.

Otherwise it's going to be tricky to have a user created quickstart repo that also has autoscaler, for example.

**Comment by marccarre** [comment]
@martina-if, they don't have to be embedded. That's just the result of me misunderstanding what we wanted, as:

- WKS does addons this way ;
- it had been mentioned in a call that we wanted something similar to fluxcd/flux/pull/2287, also similar to WKS ;
- it had been mentioned in some user stories, that we could also have `eksctl` generate these.

We may want this in the future, but apparently not now, so I'm gonna close this PR.

**Comment by martina-if** [comment]
Sorry for the misunderstanding marc!

**Pull request closed without merging**